src/components/ObservationsFlashList/MyObservationsPressable.js

```diff
@@ -23,15 +24,15 @@ const MyObservationsPressable = ( { observation, testID, children }: Props ): No
 
   const unsynced = typeof observation.wasSynced !== "undefined" && !observation.wasSynced( );
 
-  const navigateToObservation = ( ) => {
+  const navigateToObservation = debounce( ( ) => {
     const { uuid } = observation;
     if ( unsynced ) {
       prepareObsEdit( observation );
       navigateToObsEdit( navigation, setMyObsOffsetToRestore );
     } else {
       navigation.push( "ObsDetails", { uuid } );
     }
-  };
+  }, 200 );
```

Comment:
This reduces the problem but does not remove it. I can still tap quickly and get two instances of ObsDetails pushed onto the stack.